app/grandchallenge/components/views.py

```diff
@@ -453,3 +428,28 @@ def form_valid(self, form):
         for civ_set in form.cleaned_data["civ_sets_to_delete"]:
             civ_set.delete()
         return super().form_valid(form)
+
+
+class FileUploadFormFieldTemplateView(LoginRequiredMixin, View):
+
+    @property
+    def interface(self):
+        return get_object_or_404(
+            ComponentInterface, slug=self.kwargs["interface_slug"]
+        )
+
+    def get(self, request, interface_slug):
+        html_content = render_to_string(
+            UserUploadMultipleWidget.template_name,
+            {
+                "widget": UserUploadMultipleWidget().get_context(
+                    name=self.interface.slug,
+                    value=None,
+                    attrs={
+                        "id": self.interface.slug,
+                        "help_text": clean(self.interface.description),
+                    },
+                )["widget"],
+            },
+        )
+        return HttpResponse(html_content)

```

Comment:
For this new view, which only return a rendered file upload field (the actual upload is handled in the form that this field is added to), I'm unsure if we need to check permissions. This view is used in the archive item and display set forms and also in the job create form when someone chooses to upload a new noon-image file rather than selecting an existing one. Since this is only returning a form field, which is loaded on a page that is already locked down by permissions and since we check the permissions around the upload later on in the form, I think it is fine as is.

Reply:
I have added object permission checks. I check permissions for the base_object since this view is used both in create and update views, so the common denominator for object permission checking is the algorithm/reader_study/archive, not the Job/DisplaySet/ArchiveItem.